### PR TITLE
[Prim][PIR] support bce_loss op forward in prim pir

### PR DESCRIPTION
**Pull request by Eddie-Wang1120**
### PR Category

Others

### PR Types

Others

### Description

support bce_loss op forward in prim pir

### CODE REVIEW

File: test/prim/pir_prim/test_prim_sub_graph_dynamic_shape.py [hunk 1]
```diff
@@ -314,6 +318,23 @@ def setUp(self):
         self.tol = 1e-6
 
 
+class TestPrimBceLoss(TestPrimTwo):
+    def setUp(self):
+        np.random.seed(2023)
+        self.x_shape = [2, 3, 4, 5]
```
Comment (cyber-pioneer) [inline test/prim/pir_prim/test_prim_sub_graph_dynamic_shape.py]: shape调大一些

File: paddle/fluid/primitive/composite/composite.h [hunk 2]
```diff
@@ -251,6 +251,21 @@ Tensor reciprocal_decomp(const Tensor& x) {
   return full<T>(empty_shape, 1.0, x.dtype()) / x;
 }
 
+template <typename T>
+Tensor bce_loss_decomp(const Tensor& x, const Tensor& label) {
+  if (has_dynamic_shape(x.shape())) {
+    auto one = backend::full_with_tensor<T>(shape<T>(x), 1, x.dtype());
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]:
```suggestion
    auto one = full<T>(empty_shape, 1, x.dtype());
```

**Comment by cyber-pioneer** [comment]
建议调整拆解方式，合并动静态shape分支

**Comment by Eddie-Wang1120** [comment]
> 建议调整拆解方式，合并动静态shape分支

okok我试一下

**Review by cyber-pioneer — DISMISSED** [review]

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**